### DISCUSSION TITLE
Question: How to design a Pipeline for Displaying Matplotlib/Seaborn Charts in Open WebUI

### DISCUSSION THREAD

**Discussion by abhishek-ch** [open-webui/pipelines#208] [Q&A]
I want to create a pipeline that generates matplotlib or seaborn charts and displays them using Open WebUI. Currently, the Tool only supports returning strings. How can I design a pipeline that allows Open WebUI to display these plots?

**Comment by justinh-rahb** [comment]
Use Files API, upload artifacts, get URL for file, insert URL into markdown image tag ?

**Comment by abhishek-ch** [comment]
so if I used the output as markdown image, open-webui will display it ?

**Reply by ghost** [reply]
Did you figure out a way?

**Comment by HEI204** [comment]
hi, any update on it?
